File: Runtime/PatchPrioritizationCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' Patch Prioritization Collection
''' </summary>
''' <remarks></remarks>
Friend Class PatchPrioritizationCollection
    Inherits KeyedCollection(Of Integer, PatchPrioritization)

    Protected Overrides Function GetKeyForItem(item As PatchPrioritization) As Integer
        Return item.PatchPrioritizationId
    End Function

End Class

File: src/Runtime/PatchPrioritizationCollection.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2023 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class PatchPrioritizationCollection : KeyedCollection<int, PatchPrioritization>
    {
        protected override int GetKeyForItem(PatchPrioritization item)
        {
            return item.PatchPrioritizationId;
        }
    }
}
